superset/assets/javascripts/SqlLab/components/TabbedSqlEditors.jsx

```diff
@@ -28,13 +29,33 @@ let queryCount = 1;
 class TabbedSqlEditors extends React.PureComponent {
   constructor(props) {
     super(props);
-    const uri = window.location.toString();
+    const sqlLabUrl = '/superset/sqllab';
+    let query = null;
     const search = window.location.search;
-    const cleanUri = search ? uri.substring(0, uri.indexOf('?')) : uri;
-    const query = search.substring(1);
+    if (search) {
+      const queryString = search.substring(1);
+      const urlId = getParamFromQuery(queryString, 'id');
+      if (urlId) {
+        $.ajax({
+          type: 'GET',
+          async: false,
```

Comment:
I'll leave to it @ascott and @mistercrunch. 
I have fairly limited knowledge in frontend development, but it still feels wrong.